nav2_docking/opennav_docking/src/docking_server.cpp

```diff
@@ -401,6 +418,27 @@ void DockingServer::doInitialPerception(Dock * dock, geometry_msgs::msg::PoseSta
   }
 }
 
+
+
+void DockingServer::rotateToDock(const geometry_msgs::msg::PoseStamped & dock_pose){
+  rclcpp::Rate loop_rate(controller_frequency_);
+  geometry_msgs::msg::PoseStamped robot_pose = getRobotPoseInFrame(dock_pose.header.frame_id);
+  double angle_to_goal;
+  auto command = std::make_unique<geometry_msgs::msg::TwistStamped>();
+  while(rclcpp::ok()){
+    angle_to_goal = angles::shortest_angular_distance(tf2::getYaw(robot_pose.pose.orientation),
```

Comment:
A question to ask: would we rather (1) rotate purely 180 from the staging pose and have the control law handle any deviation in the staging pose or (2) rotate approximately 180, taking into account any angular offsets present in the staging pose so that its as straight-backward as possible? 

If (1), then we can just take the robot orientation and the target is just PI from that, no need to look at the dock pose.

If (2), then I think we need to look at the dock pose's orientation, not its relative position angle. But, I would agree this is up for debate about what's better. Another area @ajtudela's opinion would be appreciated :-)